apis/config/v1beta1/configuration_types.go

```diff
@@ -352,3 +355,27 @@ type ClusterQueueVisibility struct {
 	// Defaults to 10.
 	MaxCount int32 `json:"maxCount,omitempty"`
 }
+
+type PreemptionStrategy string
+
+const (
+	LessThanOrEqualToFinalShare PreemptionStrategy = "LessThanOrEqualToFinalShare"
+	LessThanInitialShare        PreemptionStrategy = "LessThanInitialShare"
```

Comment:
It's not just "medium" and "strong".

There are different semantics if you put them in different orders.

Putting `LessThanInitialShare` might lead to more preemptions. But it also gives you the most predictable behavior, because the algorithm will always try to preempt the lowest priorities first. If you only have `LessThanOrEqualToFinalShare` or you put it first, then you might end up preempting a higher priority job because the low priority one doesn't satisfy the rule.

So yes, having this enums forces the user to understand the algorithm, but anyone that is coming this way has complex use cases for fair sharing already. Otherwise, they would just be using the default.

Reply:
I agree that "medium" and "strong" would not be 100% precise. But we need to make it simpler for people to understand what knobs they have to control the behavior and what (more or less) will happen if they change them. 
The algorithm is HARD to understand and many people will get it wrong and will turn knobs in the wrong direction.